saml/src/main/java/com/linecorp/armeria/server/saml/KeyStoreCredentialResolverBuilder.java

```diff
@@ -58,20 +59,40 @@ public KeyStoreCredentialResolverBuilder(File file) {
         classLoader = null;
     }
 
+    /**
+     * Creates a builder with the file at the specified {@link Path}.
+     */
+    public KeyStoreCredentialResolverBuilder(Path path) {
+        this(requireNonNull(path, "path").toFile());
+    }
+
+    /**
+     * Creates a builder with the specified {@link ClassLoader} and {@code resourcePath}.
+     */
+    public KeyStoreCredentialResolverBuilder(ClassLoader classLoader, String resourcePath) {
+        this.classLoader = requireNonNull(classLoader, "classLoader");
+        this.resourcePath = requireNonNull(resourcePath, "resourcePath");
+        file = null;
+    }
+
     /**
      * Creates a builder with the specified {@code resourcePath} and the default {@link ClassLoader}.
+     *
+     * @deprecated Use {@link #KeyStoreCredentialResolverBuilder(ClassLoader, String)}.
      */
+    @Deprecated
     public KeyStoreCredentialResolverBuilder(String resourcePath) {
-        this(resourcePath, null);
+        this(KeyStoreCredentialResolverBuilder.class.getClassLoader(), resourcePath);
     }
 
     /**
      * Creates a builder with the specified {@code resourcePath} and {@link ClassLoader}.
+     *
+     * @deprecated Use {@link #KeyStoreCredentialResolverBuilder(ClassLoader, String)}.
      */
+    @Deprecated
     public KeyStoreCredentialResolverBuilder(String resourcePath, @Nullable ClassLoader classLoader) {
-        this.resourcePath = requireNonNull(resourcePath, "resourcePath");
-        this.classLoader = classLoader;
-        file = null;
+        this(firstNonNull(classLoader, KeyStoreCredentialResolverBuilder.class.getClassLoader()), resourcePath);
```

Comment:
nit: `classLoader != null ? classLoader : KeyStoreCredentialResolverBuilder.class.getClassLoader()` 
to avoid calling of `getClassLoader()` if `classLoader` is not null?

Reply:
Good point. Fixed